lisa/sut_orchestrator/azure/tools.py

```diff
@@ -55,3 +58,66 @@ def get_generation(self) -> str:
         else:
             generation = "1"
         return generation
+
+
+class LisDriver(Tool):
+    """
+    This is a virtual tool to detect/install LIS drivers.
+    """
+
+    __version_pattern = re.compile(r"^version:[ \t]*([^ \n]*)")
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Wget, Modinfo]
+
+    @property
+    def command(self) -> str:
+        return "modinfo hv_vmbus"
+
+    @property
+    def can_install(self) -> bool:
+        if (
+            isinstance(self.node.os, Redhat)
+            and float(self.node.os.os_version.release) < 7.8
+        ):
+            return True
+
+        return False
+
+    def _check_exists(self) -> bool:
+        if isinstance(self.node.os, Redhat):
+            # currently LIS is only supported with Redhat
+            # and its derived distros
+            if self.node.os.package_exists(
+                "kmod-microsoft-hyper-v"
+            ) and self.node.os.package_exists("microsoft-hyper-v"):
+                return True
+        return False
+
+    def _install(self) -> bool:
+        wget_tool = self.node.tools[Wget]
+        lis_path = wget_tool.get("https://aka.ms/lis", str(self.node.working_path))
+
+        result = self.node.execute(f"tar -xvzf {lis_path}")
+        if result.exit_code != 0:
+            raise LisaException(
+                "Failed to extract tar file after downloading LIS package. "
+                f"exit_code: {result.exit_code} stderr: {result.stderr}"
+            )
+        lis_folder_path = self.node.working_path.joinpath("LISISO")
+        result = self.node.execute("./install.sh", cwd=lis_folder_path)
+        if result.exit_code != 0:
+            raise LisaException(
+                f"Unable to install the LIS RPMs! exit_code: {result.exit_code}"
+                f"stderr: {result.stderr}"
+            )
+        return True
+
+    def get_version(self) -> str:
+        cmd_result = self.run()
+        if cmd_result.exit_code == 0:
+            found_version = find_patterns_in_lines(
+                cmd_result.stdout, [self.__version_pattern]
+            )
+        return found_version[0][0] if found_version[0] else ""
```

Comment:
Won't we possibly have undefined var/field problems here?

Reply:
You are right. Let me fix this.